site-src/v1alpha2/references/policy-attachment.md

```diff
@@ -0,0 +1,294 @@
+this case).
+
+![Complex Mesh Example](/images/policy/mesh-complex.png)
+
+## Target Reference API
+
+Each Policy resource MUST include a single `targetRef` field. It MUST not
+target more than one resource at a time, but it can be used to target larger
+resources such as Gateways or Namespaces that may apply to multiple child
+resources.
+
+The `targetRef` field MUST be an exact replica of the `PolicyTargetReference`
+struct included in the Gateway API. Where possible, it is recommended to use
+that struct directly instead of duplicating the type.
+
+### Policy Boilerplate
+The following structure MUST be used as for any Policy resource using this API
+pattern. Within the spec, policy resources may omit `Override` or `Default`
+fields, but at least one of them MUST be present.
+
+```go
+// ACMEServicePolicy provides a way to apply Service policy configuration with
+// the ACME implementation of the Gateway API.
+type ACMEServicePolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of ACMEServicePolicy.
+    Spec ACMEServicePolicySpec `json:"spec"`
+
+    // Status defines the current state of ACMEServicePolicy.
+    Status ACMEServicePolicyStatus `json:"status,omitempty"`
+}
+
+// ACMEServicePolicySpec defines the desired state of ACMEServicePolicy.
+type ACMEServicePolicySpec struct {
+    // TargetRef identifies an API object to apply policy to.
+    TargetRef gatewayv1a2.PolicyTargetReference `json:"targetRef"`
+
+    // Override defines policy configuration that should override policy
+    // configuration attached below the targeted resource in the hierarchy.
+    // +optional
+    Override *ACMEPolicyConfig `json:"override,omitempty"`
+
+    // Default defines default policy configuration for the targeted resource.
+    // +optional
+    Default *ACMEPolicyConfig `json:"default,omitempty"`
+}
+
+// ACMEPolicyConfig contains ACME policy configuration.
+type ACMEPolicyConfig struct {
+    // Add configurable policy here
+}
+
+// ACMEServicePolicyStatus defines the observed state of ACMEServicePolicy.
+type ACMEServicePolicyStatus struct {
+    // Conditions describe the current conditions of the ACMEServicePolicy.
+    //
+    // +optional
+    // +listType=map
+    // +listMapKey=type
+    // +kubebuilder:validation:MaxItems=8
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+```
+
+### Hierarchy
+Each policy MAY include default or override values. Overrides enable admins to
+enforce policy from the top down. Defaults enable app owners to provide default
+values from the bottom up for each individual application.
+
+![Policy Hierarchy](/images/policy/hierarchy.png)
+
+To illustrate this, consider 3 resources with the following hierarchy:
+A > B > C. When attaching the concept of defaults and overrides to that, the
+hierarchy would be expanded to this:
+
+A override > B override > C override > C default > B default > A default.
```

Comment:
I just realized that if someone has RBAC permissions to XPolicy in a namespace, they can essentially target anything in that namepace and traffic originating from that namespace.

Reply:
No action required but just noting.

Reply:
Yep, it's a different security model, and something we should be clear about. For implementations, I think it means policy resources should be designed with a coherent set of permissions in mind, and for users, it means they should consider this when determining where to place Gateways and Routes.